templates/workspace_services/azureml/terraform/main.tf

```diff
@@ -1,29 +1,61 @@
-resource "azurerm_machine_learning_workspace" "aml_workspace" {
-  name                          = local.workspace_name
-  resource_group_name           = data.azurerm_resource_group.ws.name
-  location                      = data.azurerm_resource_group.ws.location
-  application_insights_id       = azurerm_application_insights.ai.id
-  container_registry_id         = azurerm_container_registry.acr.id
-  friendly_name                 = var.display_name
-  description                   = var.description
-  high_business_impact          = true
-  key_vault_id                  = data.azurerm_key_vault.ws.id
-  public_network_access_enabled = var.is_exposed_externally ? true : false
-  storage_account_id            = azurerm_storage_account.aml.id
-  tags                          = local.tre_workspace_service_tags
-
-  identity {
-    type = "SystemAssigned"
+resource "azapi_resource" "aml_workspace" {
+  type      = "Microsoft.MachineLearningServices/workspaces@2024-10-01-preview"
```

Comment:
why not [2025-01-01-preview](https://learn.microsoft.com/en-us/azure/templates/microsoft.machinelearningservices/2025-01-01-preview/workspaces) ? or the [2024-10-01](https://learn.microsoft.com/en-us/azure/templates/microsoft.machinelearningservices/2024-10-01/workspaces) if you don't want a preview

Reply:
I need systemDatastoresAuthMode, is only in preview API, I did this before 2025-01-01-preview was out.

Will update and test.